File: src/jk_5/nailed/map/gamestart/instructions/InstructionClearInventory.java
package jk_5.nailed.map.gamestart.instructions;

import jk_5.nailed.Nailed;
import jk_5.nailed.map.gamestart.GameThread;
import jk_5.nailed.map.gamestart.IInstruction;
import jk_5.nailed.players.Player;
import jk_5.nailed.teams.Team;
import net.minecraft.src.EntityPlayerMP;

/**
 * TODO: Edit description
 *
 * @author jk-5
 */
public class InstructionClearInventory implements IInstruction {

    private String team;

    @Override
    public void injectArguments(String arguments) {
        this.team = arguments;
    }

    @Override
    public void execute(GameThread controller) throws InterruptedException {
        Team team = Nailed.teamRegistry.getTeam(this.team);
        for (Player p : Nailed.playerRegistry.getPlayers()) {
            if (p.getEntity() == null) continue;
            if (p.getTeam() == team) {
                EntityPlayerMP player = p.getEntity();
                player.inventory.clearInventory(-1, -1);
                player.inventoryContainer.detectAndSendChanges();
                if (!player.capabilities.isCreativeMode) {
                    player.updateHeldItem();
                }
            }
        }
    }
}

File: src/main/scala/jk_5/nailed/map/gameloop/instructions/InstructionClearInventory.scala
package jk_5.nailed.map.gameloop.instructions

import jk_5.nailed.map.gameloop.GameThread
import jk_5.nailed.map.gameloop.IInstruction

/**
 * No description given
 *
 * @author jk-5
 */
class InstructionClearInventory extends IInstruction {

  private var team: String = null

  def injectArguments(arguments: String) = this.team = arguments

  def execute(controller: GameThread) {
    val team = controller.getMap.getTeamManager.getTeam(this.team)
    controller.getMap.getPlayers.filter(_.getEntity.isDefined).map(_.getEntity.get).foreach(p => {
      p.inventory.clearInventory(-1, -1)
      p.inventoryContainer.detectAndSendChanges()
      if(!p.capabilities.isCreativeMode) p.updateHeldItem()
    })
  }
}
